platform/drm/cog-platform-drm.c

```diff
@@ -1499,11 +1516,27 @@ on_export_shm_buffer (void* data, struct wpe_fdo_shm_exported_buffer *exported_b
 }
 #endif
 
-gboolean
-cog_platform_plugin_setup (CogPlatform *platform,
-                           CogShell    *shell,
-                           const char  *params,
-                           GError     **error)
+static void *
+check_supported(void *data G_GNUC_UNUSED)
+{
+    if (init_drm()) {
```

Comment:
`init_drm()` is way too heavy to act as a check here.

Reply:
True, some of the code could be split and use it for the checks, instead of using the whole `init_drm()` function in the `is_supported` implementation. I've created issue #330 for that.